src/openms/source/ANALYSIS/ID/BasicProteinInferenceAlgorithm.cpp

```diff
@@ -562,15 +605,43 @@ namespace OpenMS
 
     if (group)
     {
-      //TODO you could actually also do the aggregation/inference as well as the resolution on the Graph structure
-      // but it is quite fast right now.
-      IDBoostGraph ibg{prot_run, pep_ids, 1
-                       /*static_cast<Size>(getIntOption_("nr_psms_per_spectrum"))*/, false, false};
+      //TODO you could actually also do the aggregation/inference as well as the resolution on the Graph structure.
+      // Groups would be clustered already. Saving some time.
+      // But it is quite fast right now already.
+      IDBoostGraph ibg{prot_run, pep_ids, 1, false, false};
 
       ibg.computeConnectedComponents();
-      ibg.calculateAndAnnotateIndistProteins(true);
+      if (resolve)
+      {
+        ibg.clusterIndistProteinsAndPeptides(); //TODO check in resolve or do it there if not done yet!
+        //Note: the above does not add singleton groups to graph
+        ibg.resolveGraphPeptideCentric(true);
+        ibg.annotateIndistProteins(true); // this does not really add singletons since they are not in the graph
+        IDFilter::updateProteinGroups(prot_run.getIndistinguishableProteins(), prot_run.getHits());
+        IDFilter::removeUnreferencedProteins(prot_run, pep_ids);
+        prot_run.fillIndistinguishableGroupsWithSingletons();
+      }
+      else
+      {
+        ibg.calculateAndAnnotateIndistProteins(true);
+      }
+
       auto & ipg = prot_run.getIndistinguishableProteins();
       std::sort(std::begin(ipg), std::end(ipg));
     }
+    else
+    {
+      if (resolve)
+      {
+        IDBoostGraph ibg{prot_run, pep_ids, 1, false, false};
+
+        ibg.computeConnectedComponents();
+        ibg.clusterIndistProteinsAndPeptides(); //TODO check in resolve or do it there if not done yet!
+        //Note: the above does not add singleton groups to graph
+        ibg.resolveGraphPeptideCentric(true);
```

Comment:
the code above and below looks very similar. Could you add a comment to the else branch and also to the resolve that this is without annotation or in general what the difference is e.g., to make the semantics a bit clearer